File: ReSharper.FSharp/src/Services.FSharp/src/FSharpClrLanguage.cs
using JetBrains.Application;
using JetBrains.ReSharper.Feature.Services.ClrLanguages;
using JetBrains.ReSharper.Psi;
using JetBrains.ReSharper.Psi.FSharp;

namespace JetBrains.ReSharper.Feature.Services.FSharp
{
  [ShellComponent]
  public class FSharpClrLanguage : IClrLanguagesKnown
  {
    public PsiLanguageType Language => FSharpLanguage.Instance;
  }
}

File: ReSharper.FSharp/src/FSharp.Services/FSharpClrLanguage.fs
namespace JetBrains.ReSharper.Plugins.FSharp.Services

open JetBrains.Application
open JetBrains.ReSharper.Psi
open JetBrains.ReSharper.Psi.FSharp
open JetBrains.ReSharper.Feature.Services.ClrLanguages

[<ShellComponent>]
type FSharpClrLanguage() =
    interface IClrLanguagesKnown with
        member x.Language = FSharpLanguage.Instance :> PsiLanguageType
